lib/hawkular/inventory/entities.rb

```diff
@@ -0,0 +1,217 @@
+# It contains class definitions that are used by the inventory REST client
+module Hawkular::Inventory
+  # A Basic inventory entity with id, name, path and optional properties
+  class BaseEntity
+    # @return [String] Full path of the entity
+    attr_reader :path
+    # @return [String] Name of the entity
+    attr_reader :name
+    # @return [String] Name of the entity
+    attr_reader :id
+    # @return [String] Feed this entity belongs to (or nil in case of a feedless entity)
+    attr_reader :feed
+    # @return [String] Name of the environment for this entity
+    attr_reader :env
+    # @return [String] Properties of this entity
+    attr_reader :properties
+
+    def initialize(hash)
+      @id = hash['id']
+      @path = hash['path']
+      @name = hash['name'] || @id
+      @properties = hash['properties'] || {}
+      @_hash = hash.dup
+
+      return if @path.nil?
+
+      tmp = @path.split('/')
+      tmp.each do |pair|
+        (key, val) = pair.split(';')
+        case key
+        when 'f'
+          @feed = val
+        when 'e'
+          @env = val
+        end
+      end
+    end
+
+    def ==(other)
```

Comment:
Right, it should be using `equal?` instead of `eql?` ("hash idenity"), in this case it's the same, but in general you are right.

Why is it there? I guess, the same reason as in Java.. to define where 2 objects are equal, I think in spec we sometimes fetch the entity with properties and sometimes not, and it shouldn't matter, because what makes 2 object equal is the `id` in this case